File: w09/d03/enoch_riese/mail_objects_app/spec/javascripts/package_spec.js
describe("a package", function() {
  var pack;
  beforeEach(function() {
    pack = new Package({
      contents: ["flowers", "candy"],
      fragile: false
    });
  });
  it("should have a contents array", function() {
    expect(Array.isArray(pack.contents)).toBe(true);
  });
  it("should have a fragile boolean", function() {
    expect(typeof(pack.fragile)).toBe("boolean");
  })
  describe("method: shake", function() {
    it("should break the package if it's fragile", function() {
      pack.fragile = true;
      pack.shake();
      expect(pack.broken).toBe(true);
    });
    it("should not break the package if it's not fragile", function() {
      pack.shake();
      expect(pack.broken).toBe(false);
    });
  });
  it("should tell you if it's broken with isBroken()", function() {
    expect(pack.isBroken()).toBe(false);
  });
});

File: w09/d01/matthew_korporaal/mail_object/spec/package_spec.rb
require_relative 'spec_helper'
require_relative '../package'

describe Package do

  before(:each) do
    @package = Package.new
  end

  it "should inherit from MailItem" do
    expect(Package < MailItem).to be_true
  end

  it "should have a content array" do
    expect(@package.content).to be
  end

  it "should have an is_fragile boolean" do
    expect(@package.fragile?).to be_false
  end

  it "should have an is_broken boolean" do
    expect(@package.broken?).to be_false
  end

  it "should have a method 'shake'" do
    @package.shake
    expect(@package.broken?).to be_true
  end

  it "should have a is_broken method" do
    expect(@package.broken?).to_not raise_error
  end

end